services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/service/MongoDbBasedRegistrationService.java

```diff
@@ -114,6 +95,30 @@ public void setConfig(final MongoDbBasedRegistrationConfigProperties config) {
         this.config = Objects.requireNonNull(config);
     }
 
+    /**
+     * Sets the Vert.x instance to deploy the service to.
+     *
+     * @param vertx The vertx instance.
+     * @throws NullPointerException if vertx is {@code null}.
+     */
+    @Autowired
+    public void setVertx(final Vertx vertx) {
```

Comment:
this method is not necessary, vert.x will invoke *init* with the vert.x instance and the event-loop context during deployment of the verticle. So, instead, you should implement the *init* method accordingly ...

Reply:
> So, instead, you should implement the _init_ method accordingly ...

Do you mean something like below.
```
    @Override
    public void init(final Vertx vertx, final Context context) {
        this.vertx = vertx;
    }

    @Override
    public Vertx getVertx() {
        return vertx;
    }
```

Reply:
@sophokles73 I have pushed the changes except the above one and regarding indices. Would you mind taking a look again?

Reply:
@sophokles When you said implement the _init_ method accordingly, is it as below?
 ```
@Override
    public void init(final Vertx vertx, final Context context) {
      this.vertx = Objects.requireNonNull(vertx);
    }
```
Or should I create a vertx instance with the given vertx properties inside the `init` method? 
 ```
@Override
    public void init(final Vertx vertx, final Context context) {
      this.vertx = Vertx.vertx(vertxProperties().configureVertx(new VertxOptions()));
    }
```
Would you mind giving some more insights for me?

Reply:
The former